python/tritonserver/_api/_response.py

```diff
@@ -395,14 +133,21 @@ def _from_tritonserver_inference_response(
                     name,
                     data_type,
                     shape,
-                    _data_ptr,
-                    _byte_size,
-                    _memory_type,
-                    _memory_type_id,
-                    memory_buffer,
+                    data_ptr,
+                    byte_size,
+                    memory_type,
+                    memory_type_id,
                 ) = response.output(output_index)
+                memory_buffer = MemoryBuffer(
+                    data_ptr=data_ptr,
+                    memory_type=memory_type,
+                    memory_type_id=memory_type_id,
+                    size=byte_size,
+                    owner=response,
+                )
                 tensor = Tensor(data_type, shape, memory_buffer)
-
+                if output_memory_type is not None:
+                    tensor = tensor.to_device(output_memory_type)
```

Comment:
do we need to check for error here - or will propagate?

Should check original behavior in terms of fallback behavior -

Reply:
the error will be set in the response.

it is updated to raise `InvalidArgumentError` similar to the original behavior.